docs/writing-tests/test-runner-with-vitest.mdx

```diff
@@ -0,0 +1,396 @@
+1. If you're using SvelteKit, make sure you're using `@storybook/sveltekit/vite`.
+1. Create a [test setup file](../api/portable-stories/portable-stories-vitest.mdx#setprojectannotations), `.storybook/vitest.setup.ts`.
+1. Adjust your Vitest configuration to include the plugin(s) and reference the setup file. You can reference examples, below.
+
+### Example configuration files
+
+Here are example configuration files generated by the `add` command. You can use these as a reference when setting up your project.
+
+<details>
+  <summary>Example Vitest setup file</summary>
+
+```ts title=".storybook/vitest.setup.ts"
+import { beforeAll } from 'vitest';
+// Replace your-renderer with the renderer you are using (e.g., react, vue3, svelte, etc.)
+import { setProjectAnnotations } from '@storybook/your-renderer';
+import * as projectAnnotations from './preview';
+
+const project = setProjectAnnotations(projectAnnotations);
+
+beforeAll(project.beforeAll);
+```
+</details>
+
+<details>
+  <summary>Example Vitest config file</summary>
+
+```ts title="vitest.config.ts"
+import { defineConfig, mergeConfig } from 'vitest/config'
+import viteConfig from '../vite.config'
+import { storybookTest } from '@storybook/experimental-addon-vitest/plugin'
+
+export default mergeConfig(
+  viteConfig,
+  defineConfig({
+    plugins: [
+      storybookTest({
+        storybookScript: 'yarn storybook --ci',
+      }),
+    ],
+    // Glob pattern to find story files
+    include: ['../src/**/*.stories.?(m)[jt]s?(x)'],
+    // Enable browser mode
+    browser: {
+      enabled: true,
+      name: 'chromium',
+      // Make sure to install Playwright
+      provider: 'playwright',
+      headless: true,
+    },
+    // Disabling isolation is faster and similar to how tests are isolated in Storybook itself.
+    // Consider removing this, if you have flaky tests.
+    isolate: false,
+    setupFiles: ['./.storybook/vitest.setup.ts'],
+  })
+)
+```
+</details>
+
+<details>
+  <summary>Example Vitest workspace file</summary>
+
+{/* TODO: Nextjs & SvelteKit examples */}
+```ts title="vitest.workspace.ts"
+import { defineWorkspace } from 'vitest/config'
+import { storybookTest } from '@storybook/experimental-addon-vitest/plugin'
+
+export default defineWorkspace([
+  // This is the path to your existing Vitest config files
+  './vitest.config.ts', 
+  {
+    name: 'storybook',
+    plugins: [
+      storybookTest({
+        storybookScript: 'yarn storybook --ci',
+      }),
+    ],
+    // Glob pattern to find story files
+    include: ['../src/**/*.stories.?(m)[jt]s?(x)'],
+    // Enable browser mode
+    browser: {
+      enabled: true,
+      name: 'chromium',
+      // Make sure to install Playwright
+      provider: 'playwright',
+      headless: true,
+    },
+    // Disabling isolation is faster and similar to how tests are isolated in Storybook itself.
+    // Consider removing this, if you have flaky tests.
+    isolate: false,
```

Comment:
@yannbf We should remove this, if we can not get the plugin to work with `isolate: false`